src/borg/helpers.py

```diff
@@ -1178,6 +1177,10 @@ def normpath_special(p):
             return True
         return False
 
+    @staticmethod
+    def archive_valid(archive):
+        return not archive or (archive == posixpath.normpath(archive) and archive not in ('..', '.') and '::' not in archive)
```

Comment:
what I meant is that a falsy archive name is not a valid archive name.

the only case where this is used in a overall-valid context is if we have a repo location without an archive name, but that should be handled outside of an archive_valid function.